File: pythonchallenge/find_rare_characters.py
#! /usr/bin/env python2
# find_rare_characters.py  --  created by Ing. Josef Klotzner
from collections import Counter
fobj = open("view-source_www.pythonchallenge.com_pc_def_ocr.txt", "r")
c=Counter()
for line in fobj:
    c += Counter(line.strip())
print c
"""
output:
Counter({')': 6186, '@': 6157, '(': 6154, ']': 6152, '#': 6115, '_': 6112, '[': 6108, '}': 6105, '%': 6104, '!': 6079, '+': 6066, '$': 6046, '{': 6046, '&': 6043, '*': 6034, '^': 6030, 'a': 1, 'e': 1, 'i': 1, 'l': 1, 'q': 1, 'u': 1, 't': 1, 'y': 1})
in short:   aeilquty  .... solution word in order of occurance of characters: equality
"""

File: pythonchallenge/find_rare_characters.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main()
{
	int i=0;
	int j=0;
	char* result;
	char* p;
	char* m;
	int a=0;
	int c=0;
	result=malloc(100000*sizeof(char));
	m=malloc(256*sizeof(char));
	FILE *fichier;
	fichier=fopen("view-source_www.pythonchallenge.com_pc_def_ocr.txt","r"); //enigme3.txt is the file containing the sequence of characters
	if (fichier != NULL)
	{
		p=result;
		while (!feof (fichier))
			fscanf(fichier,"%c",p++);
		fclose(fichier);
	}
	for (i=0;i<100000;i++)
	{
		a=0;
		for (j=0;j<256;j++)
		{
			if (result[i]!=m[j])
			{
				a++;
			}
			if (a==256)
			{
				m[c]=result[i];
				c++;
			}
		}
	}
	printf("%s",m);
	return 0;
}
